contracts/Template.sol

```diff
@@ -1,106 +1,129 @@
+        Vault agentOrVault = _useAgentAsVault ? _installDefaultAgentApp(dao) : _installVaultApp(dao);
+        Finance finance = _installFinanceApp(dao, agentOrVault, _financePeriod == 0 ? DEFAULT_FINANCE_PERIOD : _financePeriod);
+        TokenManager tokenManager = _installTokenManagerApp(dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        Voting voting = _installVotingApp(dao, token, _votingSettings[0], _votingSettings[1], _votingSettings[2]);
+
+        // Install placeholder-app-name
+        //--------------------------------------------------------------//
+        bytes32 _appId = keccak256(abi.encodePacked(apmNamehash("open"), keccak256("placeholder-app-name")));
+        CounterApp app = _installApp(_dao, _appId, new bytes(0), false);
+        app.initialize();
+        //--------------------------------------------------------------//
 
-        address root = msg.sender;
-        bytes32 appId = keccak256(abi.encodePacked(apmNamehash("open"), keccak256("placeholder-app-name")));
-        bytes32 votingAppId = apmNamehash("voting");
-        bytes32 tokenManagerAppId = apmNamehash("token-manager");
+        // Mint tokens
+        _mintTokens(acl, tokenManager, _holders, _stakes);
 
-        CounterApp app = CounterApp(dao.newAppInstance(appId, latestVersionAppBase(appId)));
-        Voting voting = Voting(dao.newAppInstance(votingAppId, latestVersionAppBase(votingAppId)));
-        TokenManager tokenManager = TokenManager(dao.newAppInstance(tokenManagerAppId, latestVersionAppBase(tokenManagerAppId)));
+        _setupPermissions(dao, acl, agentOrVault, voting, finance, tokenManager, app, _useAgentAsVault);
 
-        MiniMeToken token = tokenFactory.createCloneToken(MiniMeToken(0), 0, "App token", 0, "APP", true);
-        token.changeController(tokenManager);
+        _registerID(_id, dao);
+    }
 
-        app.initialize();
-        tokenManager.initialize(token, true, 0);
-        // Initialize apps
-        voting.initialize(token, 50 * PCT, 20 * PCT, 1 days);
+    function _setupPermissions(Kernel _dao, ACL _acl, Vault _agentOrVault, Voting _voting, Finance _finance, TokenManager _tokenManager, CounterApp _app, bool _useAgentAsVault) internal {
+        if (_useAgentAsVault) {
+            _createAgentPermissions(_acl, Agent(_agentOrVault), _voting, _voting);
+        }
+        _createVaultPermissions(_acl, _agentOrVault, _finance, _voting);
+        _createFinancePermissions(_acl, _finance, _voting, _voting);
+        _createEvmScriptsRegistryPermissions(_acl, _voting, _voting);
+        _createCustomVotingPermissions(_acl, _voting, _tokenManager);
+        _createCustomTokenManagerPermissions(_acl, _tokenManager, _voting);
+
+        // Setup placeholder-app-name permissions
+        //--------------------------------------------------------------//
+        _createCustomAppPermissions(_acl, _app, _voting);
+        //--------------------------------------------------------------//
```

Comment:
I don't think it's necessary to have this line between comments :)

Reply:
Cool, I thought it might help users not much proficient with solidity yet to guide them what should they edit. But most likely I underestimating users here 😅